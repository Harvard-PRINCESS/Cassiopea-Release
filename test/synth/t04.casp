letstate allmem: 32 bit 8 len 32 ref memory

letstate r: 32 bit loc

defop EQ a: 32 bit  b: 32 bit  {
   txt = "", (* "eq " ++ a.txt ++ ", " ++ b.txt, *)
   sem = [
      if a == b then
         *r <- 0x00000001
      else
         *r <- 0x00000000
   ]
}

defop SW addr: 32 bit {
   txt = "", (* "sw " ++ addr.txt, *)
   sem = [
      let ptr: 32 bit = [allmem, 0] in
      let ptr2: 32 bit = ptr b+ addr in
      store[ptr2, 32] <- *r
   ]
}
