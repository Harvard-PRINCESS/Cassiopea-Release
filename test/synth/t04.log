Command: Synthesize for spec t04.spec output to t04.out; log to t04.log.raw
Output to: t04.out
Log to: t04.log.raw
Synthesis with 0 insts starting
Found dependencies in: 0.003451s
Found value dependencies in: 0.000032s
Iter 1 Guessing...	Guessing Time: ???s
Iter 1 Checking...	Checking Time: ???s
Iter 2 Guessing...	Guessing Time: ???s
Synthesis with 0 insts failed with 1 candidates, 1 counterexamples!
Synthesis with 1 insts starting
Found dependencies in: 0.003715s
Found value dependencies in: 0.000037s
Iter 1 Guessing...	Guessing Time: ???s
Iter 1 Checking...	Checking Time: ???s
Iter 2 Guessing...	Guessing Time: ???s
Iter 2 Checking...	Checking Time: ???s
Synthesis with 1 insts succeeded! Prog:
(EQ 0b00000000000000000000000000000000 0b00000000000000000000000000000000)
	Execution Time: ???s
