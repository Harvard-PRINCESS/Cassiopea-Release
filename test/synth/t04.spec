(* github issue #159, base case *)
let ptr: 32 bit = [allmem, 0]
frame:
pre: true
post:
   *r == 0x00000001
