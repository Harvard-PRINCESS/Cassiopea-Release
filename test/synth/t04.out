(EQ 0b00000000000000000000000000000000 0b00000000000000000000000000000000)
